src/testdrive/src/action/kafka/verify.rs

```diff
@@ -167,10 +185,25 @@ impl Action for VerifyAction {
                     consumer
                         .store_offset_from_message(&message)
                         .context("storing message offset")?;
-                    actual_bytes.push((
-                        message.key().and_then(|bytes| Some(bytes.to_owned())),
-                        message.payload().and_then(|bytes| Some(bytes.to_owned())),
-                    ));
+                    let headers = self
+                        .header_keys
+                        .iter()
+                        .map(|k| {
+                            // grab the value of the first header with the matching key, if there is one.
```

Comment:
Possible weird cases here: missing header, duplicated header, non-string header, header with no value. The current code is pretty lenient, since I really have only one use-case for this so far... but we could always make it stricter now or in the future.

Reply:
Cool - I'll crank this to be maximally strict then, and future users can relax it if they want. No point designing around use-cases we don't have yet...